Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp

```diff
@@ -4246,13 +4260,56 @@ namespace ScriptCanvasEditor
             RemoveSystemTickAction(SystemTickActionFlag::CloseNextTabAction);
             CloseNextTab();
         }
+    }
 
-        if (m_systemTickActions == 0)
+    void MainWindow::ClearSaveAttempt()
+    {
+        m_saveAttemptInProgress = false;
+
+        if (!m_systemTickActions)
         {
             AZ::SystemTickBus::Handler::BusDisconnect();
         }
     }
 
+    bool MainWindow::EvaluateSaveAttempt()
+    {
+        const AZ::s64 k_saveAttemptSeconds = 20;
+
+        if (m_saveAttemptInProgress)
+        {
+            auto saveDuration = AZStd::chrono::seconds(AZStd::chrono::system_clock::now() - m_saveAttemptTime).count();
+
+            if (saveDuration > k_saveAttemptSeconds)
+            {
+                WarnOnFailedSaveAttempt();
+            }
+        }
+
+        return m_forceCloseInProgress;
+    }
+
+    void MainWindow::MarkSaveAttempt()
+    {
+        m_saveAttemptInProgress = true;
+        m_saveAttemptTime = AZStd::chrono::system_clock::now();
+
+        if (!AZ::SystemTickBus::Handler::BusIsConnected())
+        {
+            AZ::SystemTickBus::Handler::BusConnect();
+        }
+    }
+
+    void MainWindow::WarnOnFailedSaveAttempt()
+    {
+        m_forceCloseInProgress = true;
+        QMessageBox::critical(this, QString(), QObject::tr
+        ("The ScriptCanvas Editor has encountered an external bug which prevents it from tracking the file state.<br><br>"
+        "Likely the Asset Processor has crashed. ScriptCanvas files may have saved successfully, but the O3DE Engine and Asset Processor should be restarted before continuing work."));
```

Comment:
Perhaps avoid speculation, instead just direct users to a solution "Verify that the Asset Processor is running". Ideally, users can be prompted to launch it right from the dialog as we do when an AP disconnection is detected (i.e. `MainWindow::ShowConnectionDisconnectedDialog()`), it provides a better customer experience.
In fact, the messaging could mirror the one there:
```
QMessageBox::critical(this, tr("Asset Processor has disconnected."),
            tr("Asset Processor is not connected. Please try (re)starting the asset processor or restarting the Editor.<br><br>"
                "Data may be lost while the asset processor is not running!<br>"
                "The status of the asset processor can be monitored from the editor in the bottom-right corner of the status bar."));
```

Reply:
Launching AP to a running state will not restore the SC main window to a usable, correct, or reliable state. Even re-launching the SC window does not put it in a reliable state once the AP has crashed, at least while the SC editor window was waiting for save-related callbacks.

Reply:
It would still close the SC window, but it gives the user the ability to relaunch, just the window and not have to dig around for assetprocessor.exe